crates/matrix-sdk-base/src/event_cache/store/memory_store.rs

```diff
@@ -148,6 +148,35 @@ impl EventCacheStore for MemoryStore {
         Ok(())
     }
 
+    async fn filter_duplicated_events(
+        &self,
+        room_id: &RoomId,
+        mut events: Vec<OwnedEventId>,
+    ) -> Result<Vec<OwnedEventId>, Self::Error> {
+        // Collect all duplicated events.
+        let inner = self.inner.read().unwrap();
+
+        let mut duplicated_events = Vec::new();
+
+        for event in inner.events.unordered_events(room_id) {
+            // If `events` is empty, we can short-circuit.
+            if events.is_empty() {
+                break;
+            }
```

Comment:
Can you `if events.is_empty() { return; }` at the top of the events, instead?

Reply:
No because we remove events one by one from `events`. When it's empty, we can stop iterating all other events.